git-repository/examples/new.rs

```diff
@@ -1,35 +1,33 @@
 // cargo run -p git-repository --example new
 
 use git_repository as git;
-use std::fs;
-use std::path::Path;
-
-const NEW_REPO_PATH: &str = "example-new-repo";
 
 fn main() -> Result<(), Box<dyn std::error::Error>> {
-    let path = Path::new(NEW_REPO_PATH);
-    if path.exists() {
-        fs::remove_dir_all(path)?;
-    }
-    let repo = git::init(path)?.apply_environment();
+    let tmp = tempfile::TempDir::new()?;
+    let work_dir = tmp.path().join("repo-non-bare");
+    let repo = git::init(work_dir)?;
+
     println!(
-        "Repo: {}",
-        repo.work_dir().as_deref().unwrap_or(repo.git_dir()).display()
+        "Repo: {:?}",
+        repo.work_dir().expect("non-bare repositories have a work-dir")
     );
-    let empty_tree_id = repo.write_object(&git::objs::Tree::empty()).unwrap().detach();
-    let author = git::actor::Signature {
+
+    let empty_tree_id = repo.write_object(git::objs::Tree::empty())?;
+    let author = git::actor::SignatureRef {
         name: "Maria Sanchez".into(),
         email: "maria@example.com".into(),
         time: git_date::Time::now_local_or_utc(),
     };
     let id = repo.commit(
         "HEAD",
-        author.to_ref(),
-        author.to_ref(),
+        author,
+        author,
         "initial commit",
         empty_tree_id,
         git::commit::NO_PARENT_IDS,
     )?;
-    println!("new object id: {:?}", id);
+    println!("new commit id with empty tree: {:?}", id);
+
+    tmp.into_path();
```

Comment:
wondering whether `tmp.into_path();` is serving a purpose here... or unfinished?

Reply:
That's a great point! I tried to make that clearer with [e540a76](https://github.com/Byron/gitoxide/pull/491/commits/e540a763a3e08bba669652116cb696ab51ebc21f).

Generally I want to refrain from deleting directories on the users disk assuming they are owned by us, and using temporary directories that are persisted for inspection is one way around that.

I could also imagine to take input from the user to point to an empty directory, but that would generally mean more checks to be performed or more complex code which doesn't really contribute to the intent of the example itself.

Reply:
um... that doesn't make it clear (to me).  From rust docs:
> The [TempDir](https://docs.rs/tempdir/latest/tempdir/struct.TempDir.html) type creates a directory on the file system that is deleted once it goes out of scope.

so... with your change `_persisted` is still a local variable that goes out of scope when main() ends, why isn't the directory deleted?

Reply:
Thanks for the feedback! I have just updated it to another version which I actually prefer. It resembles what app developers would do and shows off typical error handling as well. All this is safe to use as well and should overall be much easier to understand.

What do you think?